src/components/PostDetails.tsx

```diff
@@ -1,105 +1,64 @@
 import React, { useEffect, useState } from 'react';
+import * as commentsApi from '../api/comments';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addComment,
+  fetchComments,
+  removeComment,
+  setError,
+} from '../features/comments';
+import { CommentData } from '../types/Comment';
+
+export const PostDetails: React.FC = () => {
+  const {
+    loaded,
+    hasError,
+    comments,
+  } = useAppSelector(state => state.comments);
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const dispatch = useAppDispatch();
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  useEffect(() => {
+    if (selectedPost?.id) {
+      dispatch(fetchComments(selectedPost?.id));
+    }
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
     setVisible(false);
-    setError(false);
+  }, [dispatch, selectedPost?.id]);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
+  const deleteComment = async (commentId: number) => {
+    dispatch(removeComment(commentId));
 
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
+    await commentsApi.deleteComment(commentId);
   };
 
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
-
-  const addComment = async ({ name, email, body }: CommentData) => {
+  const createComment = async ({ name, email, body }: CommentData) => {
     try {
       const newComment = await commentsApi.createComment({
         name,
         email,
         body,
-        postId: post.id,
+        postId: selectedPost?.id,
```

Comment:
I think it's better to check if some post is selected here before creating comments